sci_analysis/graphs/vector.py

```diff
@@ -262,19 +263,32 @@ def __init__(self, xdata, ydata=None, **kwargs):
         :param _points: Display the scatter points.
         :param _contours: Display the density contours
         :param _boxplot_borders: Display the boxplot borders
+        :param _highlight: an array-like with points to highlight based on labels
+        :param _labels: a vector object with the graph labels
         :param _title: The title of the graph.
         :param _save_to: Save the graph to the specified path.
         :return: pass
         """
         self._fit = kwargs['fit'] if 'fit' in kwargs else True
         self._points = kwargs['points'] if 'points' in kwargs else True
+        self._labels = kwargs['labels'] if 'labels' in kwargs else [False]
+        self._highlight = kwargs['highlight'] if 'highlight' in kwargs else None
         self._contours = kwargs['contours'] if 'contours' in kwargs else False
         self._contour_props = (31, 1.1)
         self._boxplot_borders = kwargs['boxplot_borders'] if 'boxplot_borders' in kwargs else False
         self._title = kwargs['title'] if 'title' in kwargs else 'Bivariate'
         self._save_to = kwargs['save_to'] if 'save_to' in kwargs else None
         yname = kwargs['yname'] if 'yname' in kwargs else 'y Data'
         xname = kwargs['xname'] if 'xname' in kwargs else 'x Data'
+        if 'labels' in kwargs:
+            #filter out values if drop_nan was used
+            self._labels = self._labels[~xdata._dropped_vals]
```

Comment:
xdata might or might not be a Vector object at this point in the constructor. If it isn't a Vector object, you'll get an AttributeError which is why your unit tests are failing. I think you can fix this by moving these label and highlight setup lines to after the call to super(), or by moving them into draw().

Good job with the unit tests by the way. The silver lining with them failing at the moment is they caught a real use case that should be addressed, and will result in more robust code.